lib/stack_master/stack_definition.rb

```diff
@@ -41,6 +41,36 @@ def template_dir
     end
 
     def template_file_path
+
+      # Download S3 template if you have skipped upload
+      # This is used for diff/validate commands
+      if s3_configured? && s3['use_remote']
+
+        StackMaster.stdout.puts "I see use_remote flag. Templates will be referenced from S3 bucket and not locally"
+        s3_obj = ::Aws::S3::Client.new(region: s3['region'])
+        raise "Unable to Get S3 driver for downloading template" if s3_obj.nil?
+
+        # Construct file temporary file location for downloading
+        # template from S3 bucket
+        _template_dir = Dir.tmpdir()
+        _template_file_path = File.join(_template_dir, template)
```

Comment:
Don't have to but it's a good indication in the code to show variables are temporary